core/blockchain.go

```diff
@@ -2498,3 +2505,19 @@ func (bc *BlockChain) SetBlockValidatorAndProcessorForTesting(v Validator, p Pro
 func (bc *BlockChain) SetTrieFlushInterval(interval time.Duration) {
 	bc.flushInterval.Store(int64(interval))
 }
+
+// RealizedSecondsPerBlock computes the average seconds per block for the last n blocks
+// if n <= 0 the BlocksPerSecondCheckLength is used
+func (bc *BlockChain) RealizedSecondsPerBlock(n int64) (float64, error) {
+	if n <= 0 {
+		n = BlocksPerSecondCheckLength
+	}
+	curBlockNum := bc.CurrentHeader().Number.Int64()
+	prevBlockNum := math.Max64(0, curBlockNum-n)
```

Comment:
Ironically, since you have to convert to `float64` anyway, this function could also use the standard library `math.Max` :)